src/modules/logger/logger.cpp

```diff
@@ -1122,6 +1126,10 @@ bool Logger::start_stop_logging()
 
 	// only start/stop if this is a state transition
 	if (updated && _prev_state != desired_state) {
+		if (_prev_state && _log_mode == LogMode::arm_until_shutdown) {
+			return true;
+		}
```

Comment:
Can you move this logic to https://github.com/PX4/PX4-Autopilot/blob/5bff4f19e44a455968db381bf33d78a75223cfb8/src/modules/logger/logger.cpp#L1120 and update `desired_state` instead:
`desired_state = (vehicle_status.arming_state == vehicle_status_s::ARMING_STATE_ARMED) || (_prev_state && _log_mode == LogMode::arm_until_shutdown);`

Reply:
I agree.